packages/monte_carlo/event/population_control/src/MonteCarlo_Importance.cpp

```diff
@@ -0,0 +1,76 @@
+//---------------------------------------------------------------------------//
+//!
+//! \file   MonteCarlo_WeightWindowMesh.cpp
+//! \author Philip Britt
+//! \brief  Weight window mesh class definition
+//!
+//---------------------------------------------------------------------------//
+
+// std includes
+#include <math.h>
+
+// FRENSIE Includes
+#include "FRENSIE_Archives.hpp"
+#include "MonteCarlo_Importance.hpp"
+#include "Utility_ExceptionTestMacros.hpp"
+#include "Utility_RandomNumberGenerator.hpp"
+
+namespace MonteCarlo{
+
+// Update the particle state and bank
+void Importance::checkParticleWithPopulationController( ParticleState& particle,
+                                                              ParticleBank& bank ) const
+{
+
+  //! Make sure there is a weight window where this particle is.
+  if(this->isParticleInImportanceDiscretization( particle ))
+  {
+    // Apply importances
+    if(particle.getCollisionNumber() >= 1)
+    {
+      if(particle.getCollisionNumber() > 1)
+      {
+        particle.updateImportance(this->getImportance(particle));
+      }else if(particle.getCollisionNumber() == 1)
```

Comment:
```suggestion
      }else
```
as `particle.getCollisionNumber() >= 1` is already tested